File: archive/perfil-nativo/android/ElevationProfile.kt
package app.calendariociclismo.android.data.model

import kotlinx.serialization.Serializable

@Serializable
data class ElevationProfile(
    val distance: Double,
    val elevationGain: Int? = null,
    val elevationLoss: Int? = null,
    val minElevation: Int? = null,
    val maxElevation: Int? = null,
    val points: List<ElevationPoint> = emptyList(),
)

@Serializable
data class ElevationPoint(val km: Double, val alt: Int)

@Serializable
data class ProfileSummit(
    val name: String? = null,
    val km: Double,
    val altitude: Int? = null,
    val category: String? = null,
    val side: String? = null,
)

@Serializable
data class ProfileWaypoint(
    val name: String? = null,
    val km: Double,
    val type: String,
    val lengthKm: Double? = null,
)

File: archive/perfil-nativo/ios/ElevationProfile.swift
import Foundation

struct ElevationProfile: Codable {
    let distance: Double
    let elevationGain: Int?
    let elevationLoss: Int?
    let minElevation: Int?
    let maxElevation: Int?
    let points: [ElevationPoint]
}

struct ElevationPoint: Codable {
    let km: Double
    let alt: Int
}

struct ProfileSummit: Codable, Identifiable {
    var id: String { "\(km)-\(name ?? "")" }
    let name: String?
    let km: Double
    let altitude: Int?
    let category: String?
    let side: String?
}

struct ProfileWaypoint: Codable, Identifiable {
    var id: String { "\(km)-\(type)" }
    let name: String?
    let km: Double
    let type: String
    let lengthKm: Double?
}
